l10n_it_fatturapa_out/models/account.py

```diff
@@ -49,6 +49,12 @@ def _compute_fatturapa_state(self):
 
     def preventive_checks(self):
         for invoice in self:
+            for aml in invoice.line_ids:
```

Comment:
@TheMule71  qui non dovresti controllare invoice_line_ids invece di line_ids ?

Reply:
Cambia poco, 
```                if aml.product_id and not aml.tax_ids:
                    raise UserError(
                        _("Invoice %s contains product lines w/o taxes") % invoice.name
                    )
```
in ogni caso seleziona le righe con aml.product_id non nullo.

In teoria, invoice_line_ids è line_ids con un filtro applicato, exclude_from_invoice_tab == False.  exclude_from_invoice_tab è (per definizione) solo un campo tecnico che esclude la visualizzazione di alcune righe dalla fattura. Certo se poi segui il codice ovviamente certe righe vengono escluse per motivi di business logic, ma rimane solo un campo tecnico che governa la visibilità o meno.

Comunque il codice di fatto è equivalente, se posso renderlo più leggibile lo faccio volentieri...

Reply:
si sarebbe sicuramente più leggibile e punterebbe alle vere righe della fattura.

Reply:
@matteoopenf nel frattempo ho mergiato visto e considerato che la tua richiesta è stata realizzata.